src/Rules/Between.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spotwilliams\Validator\Rules;
+
+use Spotwilliams\Validator\Exceptions\ValidationFailed;
+use Spotwilliams\Validator\Contracts\ValidationRule;
+
+class Between implements ValidationRule
+{
+    private $field;
+    private $value;
+    private $constraint;
+
+    /** @inheritDoc */
+    public function apply(string $ruleName, string $field, mixed $value = null, $constraint = []): bool
+    {
+        $this->field = $field;
+        $this->value = $value;
+        $this->constraint = $constraint;
+
+        $this->isValidFieldAndConstraints();
+
+        return true;
```

Comment:
Hi @cacing69 thanks for your changes. I believe I didn't explain myself clear enough. 

My idea it was to separate the lines that are dedicated to check the goodness of the parameters from the lines that are dedicated to apply the rule per se. So, for example:

```php
public function apply(string $ruleName, string $field, mixed $value = null, $constraint = []): bool
    {
        $this->field = $field;
        $this->value = $value;
        $this->constraint = $constraint;

        $this->validateParameters();

if($this->valueLessThan($this->constraint["min"])) {
            throw new ValidationFailed("The field `{$this->field}` value is too low, allowed range is between {$this->constraint["min"]}-{$this->constraint["max"]}.");
        }

        if($this->valueGreaterThan($this->constraint["max"])) {
            throw new ValidationFailed("The field `{$this->field}` value is too high, allowed range is between {$this->constraint["min"]}-{$this->constraint["max"]}.");
        }

        return true;

}

```

By doing this you can isolate what lines are checking the parameters that you're receiving from the developer (rule configuration) from those that are coming from the user (the real validation)

In this case I discourage to add the received paremeters in properties in the class because now in the new `validateParameters` method it's better to see with clarity what you're receiving (what you expect actually)